test/e2e_test.go

```diff
@@ -120,6 +135,22 @@ var _ = Describe("MetricsServer", func() {
 		usage := ms.Containers[0].Usage
 		Expect(usage.Cpu().MilliValue()).To(BeNumerically("~", 50, 10), "Unexpected value of cpu")
 	})
+	It("returns accurate CPU metric", func() {
+		Expect(err).NotTo(HaveOccurred(), "Failed to create %q pod", cpuConsumerWithInitContainerPodName)
+		deadline := time.Now().Add(60 * time.Second)
+		var ms *v1beta1.PodMetrics
+		for {
+			ms, err = mclient.MetricsV1beta1().PodMetricses(metav1.NamespaceDefault).Get(context.TODO(), cpuConsumerWithInitContainerPodName, metav1.GetOptions{})
+			if err == nil || time.Now().After(deadline) {
+				break
+			}
+			time.Sleep(5 * time.Second)
+		}
+		Expect(err).NotTo(HaveOccurred(), "Failed to get %q pod", cpuConsumerWithInitContainerPodName)
+		Expect(ms.Containers).To(HaveLen(1), "Unexpected number of containers")
+		usage := ms.Containers[0].Usage
```

Comment:
Let's validate name of container here, it's important to distinquish if that's a init container or not